src/mito2/src/read/compat.rs

```diff
@@ -255,8 +271,21 @@ fn may_compat_fields(
         .iter()
         .map(|column_id| {
             if let Some(index) = source_field_index.get(column_id) {
+                // TODO(kould): more Modify case, e.g. column name
```

Comment:
We use column id internally so modifying the column name should not affect this.